addons/stock/models/stock_warehouse.py

```diff
@@ -490,12 +490,14 @@ def _create_missing_locations(self, vals):
         avoid mistakes during picking types and rules creation.
         """
         for warehouse in self:
+            company_id = vals.get('company_id', warehouse.company_id.id)
+            vals.update(company_id=company_id)
```

Comment:
This is wrong.

You should not update `vals`. You are in a loop. This means that in the second pass, the `company_id` is already set in `vals` (and this is the company of the previous warehouse, not our).

This is also problematic as this method is called by `write`, which mean writing any value on multiple warehouses will overwrite the company on all ([it's the case in saas-12.3](https://github.com/odoo/odoo/blob/9bf0b04adb4e19743b30e7d3bdb60462ff8d4af4/addons/mrp/data/mrp_demo.xml#L727-L730)).